lara-typescript/src/interactive-api-client/index.ts

```diff
@@ -0,0 +1,344 @@
+
+    constructor(options: IClientOptions<AuthoredState, InteractiveState, GlobalInteractiveState>) {
+      options.verbosity = Client.GetLocalVerbosityLevel(options.name) || options.verbosity;
+      this.options = options;
+      this.log("debug", "constructor: verbosity level is", options.verbosity);
+
+      if (inIframe) {
+        this.log("debug", "constructor: in iframe");
+        if (!options.startDisconnected) {
+          this.log("debug", "constructor: automatically connecting");
+          this.connect();
+        }
+      } else {
+        this.log("debug", "constructor: not in iframe");
+      }
+    }
+
+    public connect() {
+      if (inIframe) {
+        if (!this.phone) {
+          this.log("debug", "#connect: connecting");
+          this.phone = iframePhone.getIFrameEndpoint();
+
+          this.addListener("hello", () => {
+            if (this.options.onHello) {
+              this.log("debug", "onHello listener called, calling client callback");
+              this.options.onHello();
+            } else {
+              this.log("debug", "onHello listener called, no client callback found");
+            }
+
+            if (this.options.supportedFeatures) {
+              this.setSupportedFeatures(this.options.supportedFeatures);
+            }
+          });
+
+          this.addListener("getLearnerUrl", () => {
+            if (this.options.onGetLearnerUrl) {
+              const learnerUrl = this.options.onGetLearnerUrl();
+              this.log("debug", "getLearnerUrl listener called, client returned:", learnerUrl);
+              this.setLearnerUrl(learnerUrl);
+            } else {
+              this.log("debug", "getLearnerUrl listener called, no client callback found");
+            }
+          });
+
+          this.addListener("getInteractiveState", () => {
+            if (this.options.onGetInteractiveState) {
+              const interactiveState = this.options.onGetInteractiveState();
+              this.log("debug", "getInteractiveState listener called, client returned:", interactiveState);
+              this.setInteractiveState(interactiveState);
+            } else {
+              this.log("debug", "getInteractiveState listener called, no client callback found");
+            }
+          });
+
+          this.addListener("initInteractive",
+            // tslint:disable-next-line:max-line-length
+            (initMessage: LaraInteractiveApi.IInitInteractive<AuthoredState, InteractiveState, GlobalInteractiveState>) => {
+              if (this.options.onInitInteractive) {
+                this.log("debug", "initInteractive listener called, calling client callback");
+                this.options.onInitInteractive(initMessage);
+              } else {
+                this.log("debug", "initInteractive listener called, no client callback found");
+              }
+          });
+
+          this.addListener("loadInteractive", (interactiveState: InteractiveState | null) => {
```

Comment:
This is deprecated. I think it should be removed from the client. If an interactive is upgraded to use the client, then it should also be upgraded to no use deprecated messages like this.